### ISSUE TITLE
Automatic fallback to AMQP over WebSocket doesn't seem to happen

### ISSUE THREAD

**Issue by anhashia** [Azure/azure-iot-sdk-csharp#385]
- **OS and version used:** <VERSION>  Windows 10

- **SDK version used:** <VERSION> 

# Description of the issue:
If AMQP connection times out or is blocked for some reason, the SDK should automatically fall back to AMQP over WebSocket. Automatic fall back doesn't seem to work and connection times out.

# Code sample exhibiting the issue:
````
DeviceClient deviceClient = DeviceClient.CreateFromConnectionString(DeviceConnectionString, new       ITransportSettings[]
                        {
                            new AmqpTransportSettings(TransportType.Amqp_Tcp_Only),
                            new AmqpTransportSettings(TransportType.Amqp_WebSocket_Only)
                                                        
                        }
                );
              
````
# Console log of the issue:

**Comment by CIPop** [comment]
Thanks for reporting @anhashia !

**Comment by talmdal** [comment]
Just for completeness... this also occurs on macOS 10.13.5

**Comment by anhashia** [comment]
If port 5671 is blocked from get go, then the traffic does falls back to **443**. Tested on recently released  **1.20.0**

**Comment by az-iot-builder-01** [comment]
@CIPop, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)
